crates/aesgcm/src/ffi/aesgcm.rs

```diff
@@ -0,0 +1,91 @@
+use std::ffi::{c_char, c_int, c_uchar};
+
+use cosmian_ffi_utils::{ffi_read_bytes, ffi_unwrap, ffi_write_bytes};
+
+use crate::core::{ReExposedAesGcm, KEY_LENGTH, NONCE_LENGTH};
+
+unsafe extern "C" fn aesgcm(
+    output_ptr: *mut c_uchar,
+    output_len: *mut c_int,
+    input_data_ptr: *const c_char,
+    input_data_len: c_int,
+    key_ptr: *const c_char,
+    key_len: c_int,
+    nonce_ptr: *const c_char,
+    nonce_len: c_int,
+    encrypt_flag: bool,
+) -> c_int {
+    let input_data_bytes = ffi_read_bytes!("input_data", input_data_ptr, input_data_len);
+    let key_bytes = ffi_read_bytes!("key", key_ptr, key_len);
+    let key: [u8; KEY_LENGTH] = ffi_unwrap!(
+        key_bytes.try_into(),
+        "AESGCM invalid key length, expected {KEY_LENGTH:?}"
+    );
```

Comment:
Keys should be zeroized. Is is possible to zeroize the key bytes pointed by `key_ptr`? Maybe we will need to rewrite the `ffi_read_bytes!` macros in order to take the destination as input like we did for the hashing macros in crypto_core.